### PR TITLE
Connector Wizard

### PR DESCRIPTION
**Pull request by LucGenetier**
Implement connector GetParameter APIs containing dynamic Intellisence capabilities

### CODE REVIEW

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.ConnectorFunction.#ctor(Microsoft.OpenApi.Models.OpenApiOperation,System.String,System.String,System.Net.Http.HttpMethod)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.ConnectorParameter.#ctor(System.String,Microsoft.PowerFx.Types.FormulaType,System.String,Microsoft.PowerFx.Types.FormulaValue)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiParser.GetFunctions(Microsoft.OpenApi.Models.OpenApiDocument)

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.ConnectorFunction.#ctor(Microsoft.OpenApi.Models.OpenApiOperation,System.String,System.String,System.Net.Http.HttpMethod)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.ConnectorParameter.#ctor(System.String,Microsoft.PowerFx.Types.FormulaType,System.String,Microsoft.PowerFx.Types.FormulaValue)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiParser.GetFunctions(Microsoft.OpenApi.Models.OpenApiDocument)

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/ConnectorSuggestions.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.PowerFx.Types
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Types/ConnectorSuggestions.cs:6]: can we move to Microsoft.PowerFx.Intellisense?

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs [hunk 2]
```diff
@@ -356,14 +358,77 @@ public static void AddTopLevelSuggestionsForCursorType(IntellisenseData.Intellis
...
-                        AddSuggestion(intellisenseData, suggestion, SuggestionKind.Global, SuggestionIconKind.Other, DType.String, false);
+                        // for all parameters, except last one we have direct parameters / not in a record
+                        if (argPosition < info.Function.MaxArity - 1)
+                        {
+                            string sg = suggestion.Suggestion switch
+                            {
+                                StringValue sv => sv.Value,
+                                NumberValue nv => nv.Value.ToString(CultureInfo.InvariantCulture),
+                                _ => null
+                            };
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs:383]: FormulaValue.ToExpression

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs [hunk 3]
```diff
@@ -356,14 +358,77 @@ public static void AddTopLevelSuggestionsForCursorType(IntellisenseData.Intellis
...
+
+                            if (!string.IsNullOrEmpty(sg))
+                            {
+                                AddSuggestion(intellisenseData, $@"""{sg}""", SuggestionKind.Global, SuggestionIconKind.Other, DType.String, false);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs:387]: >$@"""{sg}""" [](http://example.com/codeflow?start=64&length=12)

And FormulaValue.ToExpression will give us the quotes for stirngs.

Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs:387]: >DType.String [](http://example.com/codeflow?start=127&length=12)

sv.Type?

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.ConnectorFunction.#ctor(Microsoft.OpenApi.Models.OpenApiOperation,System.String,System.String,System.Net.Http.HttpMethod)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.ConnectorParameter.#ctor(System.String,Microsoft.PowerFx.Types.FormulaType,System.String,Microsoft.PowerFx.Types.FormulaValue)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiParser.GetFunctions(Microsoft.OpenApi.Models.OpenApiDocument)

**Pull request merged**